### PR TITLE
Component/List Box

### PR DESCRIPTION
**Pull request by haild173094**

### CODE REVIEW

File: src/components/Listbox/components/TextOption/TextOption.vue [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+interface TextOptionProps {
+  // Whether the option is selected
+  selected?: boolean;
+  // Whether the option is disabled
+  disabled?: boolean;
+}
+
+const props = defineProps<TextOptionProps>();
+
+const { allowMultiple } = inject('combobox-list-option-context', { allowMultiple: false });
```
Comment (juzser) [inline src/components/Listbox/components/TextOption/TextOption.vue:50]: Put this as service in `@/use`, define type for the context

File: src/components/Listbox/components/TextOption/TextOption.vue [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+const isAction = inject('action-context', false);
```
Comment (juzser) [inline src/components/Listbox/components/TextOption/TextOption.vue:51]: Is it a const or Ref?

File: src/components/Listbox/components/Action/Action.vue [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  selected?: boolean;
+  // Option is disabled
+  disabled?: boolean;
+  // Adds a border-bottom to the Option
+  divider?: boolean;
+  // Display icon inside Action
+  icon?: IconSource;
+}
+
+provide('actionContext', true);
```
Comment (juzser) [inline src/components/Listbox/components/Action/Action.vue:33]: The key is different between provide & inject underneath.

**Review by juzser — CHANGES_REQUESTED** [review]

File: src/components/Listbox/components/TextOption/TextOption.vue (hunk 2 above)
Comment (haild173094) [reply src/components/Listbox/components/TextOption/TextOption.vue]: It's const when parent provide('action', true);

**Review by juzser — APPROVED** [review]

**Pull request merged**